File: Semestrul 4/MPP/Proiecte/java_gui/src/main/java/map/service/CazService.java
package map.service;

import map.domain.Caz;
import map.domain.CazDTO;
import map.repository.database.CazDBRepo;
import map.repository.database.CazRepo0;

import java.util.ArrayList;
import java.util.List;

public class CazService {

    private CazRepo0 cazDBRepo;

    public CazService(CazDBRepo cazDBRepo) {
        this.cazDBRepo = cazDBRepo;
    }

    public List<Caz> findAll() {
        return (List<Caz>) cazDBRepo.findAll();
    }

    public List<CazDTO> createCazDTOList(List<Caz> cazuri, DonatieService donatieService) {
        if (cazuri == null || cazuri.isEmpty()) {
            return List.of();
        }

        List<CazDTO> cazDTOList = new ArrayList<>();
        for (Caz caz : cazuri) {
            int sumaDonatii = donatieService.getSumaDonatiiPentruCaz(caz.getId());

            CazDTO dto = new CazDTO(caz.getNume_caz(), sumaDonatii);
            cazDTOList.add(dto);
        }
        return cazDTOList;
    }

    public Caz findById(Integer id) {
        return null;
    }
}

File: Semestrul 4/MPP/Proiecte/c#_gui/service/CazService.cs
using System.Collections.Generic;
using csharp_avalonia.domain;
using csharp_avalonia.repository;
using csharp_avalonia.service;

public class CazService
{
    private readonly CazRepo0 cazRepo;

    public CazService(CazRepo0 cazRepo1)
    {
        cazRepo = cazRepo1;
    }

    public IEnumerable<Caz> FindAll()
    {
        return cazRepo.findAll();
    }

    public List<CazDTO> CreateCazDTOList(IEnumerable<Caz> cazuri, DonatieService donatieService)
    {
        var dtoList = new List<CazDTO>();
        foreach (var caz in cazuri)
        {
            int sumaDonatii = donatieService.GetSumaDonatiiPentruCaz(caz.Id);
            dtoList.Add(new CazDTO(caz.Nume_caz, sumaDonatii));
        }

        return dtoList;
    }
}

